src/vstart.sh

```diff
@@ -340,7 +340,7 @@ if [ "$overwrite_conf" -eq 0 ]; then
 else
     if [ "$new" -ne 0 ]; then
         # only delete if -n
-        asok_dir=`dirname $($CEPH_BIN/ceph-conf --show-config-value admin_socket)`
+        asok_dir=`dirname $($CEPH_BIN/ceph-conf -c $conf_fn --show-config-value admin_socket)`
```

Comment:
it looks like this is breaking `vstart.sh -n` when no there's ceph.conf:
> ~/ceph/build $ MON=1 OSD=1 RGW=1 MDS=0 MGR=0 ../src/vstart.sh -n -d --rgw_frontend beast
global_init: unable to open config file from search list /home/cbodley/ceph/build/ceph.conf
dirname: missing operand
Try 'dirname --help' for more information.

Reply:
If ceph.conf didn't exist.
./bin/ceph-conf  --show-config-value admin_socket
did not load config file, using default settings.
/var/run/ceph/ceph-client.admin.asok

 ./bin/ceph-conf  -c ceph.conf --show-config-value admin_socket
global_init: unable to open config file from search list ceph.conf

For this case, with -n. We shouldn't  specify the ceph.conf path.

Reply:
see #21019